### ISSUE TITLE
Embedding python simple example uses deprecated function and uses a function improperly

### ISSUE THREAD

**Issue by heatherg1313** [python/cpython#103138]
When compiling the "Very High Level Embedding" example from
https://docs.python.org/3/extending/embedding.html

First I get this warning, indicating that the "optional but recommended" function is deprecated. Why is the documentation recommending a deprecated function?

```
main.c: In function 'main':
main.c:12:5: warning: 'Py_SetProgramName' is deprecated [-Wdeprecated-declarations]
   12 |     Py_SetProgramName(program);  /* optional but recommended */
      |     ^~~~~~~~~~~~~~~~~
In file included from /usr/include/python3.11/Python.h:94,
                 from main.c:2:
/usr/include/python3.11/pylifecycle.h:37:38: note: declared here
   37 | Py_DEPRECATED(3.11) PyAPI_FUNC(void) Py_SetProgramName(const wchar_t *);
      |                                      ^~~~~~~~~~~~~~~~~
```

Secondly, the very first line of main() in this example calls Py_DecodeLocale(), but the documentation for that function explicitly states "This function must not be called before Python is preinitialized and so that the LC_CTYPE locale is properly configured: see the Py_PreInitialize() function.". In other words, the example code does something that the documentation says should not be done, which is setting a bad example for those trying to learn how to use the API.

**Comment by sunmy2019** [comment]
This piece of code was written 9 years ago. Things changed a lot.

See https://github.com/python/cpython/commit/25e014bd91e2f64b3231ef2bf5ddac8bb99ed637
 
You should use the newer API. https://docs.python.org/3/c-api/init_config.html#c.PyConfig.program_name

**Comment by heatherg1313** [comment]
Then the example should be updated to use the newer API, particularly since a user who's trying to learn how to embed Python will likely see this example first, and will likely try to emulate it.

**Comment by lebigmac1** [comment]
Hi. I hope I have come to the right place. Can someone please help me compile a basic python script to a binary that I can run on my phone (aarch64 architecture) ? Thanks. I've been trying for a whole week now without luck :/

**Comment by JosephSBoyle** [comment]
Hi @lebigmac1, this repo is for development of Python, I think your question would be better suited to a site like StackOverflow!:)

**Comment by AlexWaygood** [comment]
> Hi @lebigmac1, this repo is for development of Python, I think your question would be better suited to a site like StackOverflow!:)

Yup, StackOverflow or https://discuss.python.org/c/users/7 would be a better place to ask this question :)
